fix: testPlayerConditionToggleShowsAndHidesInStripRow fails on iOS 26 — player strip row not found in AX tree

## Summary

\`testPlayerConditionToggleShowsAndHidesInStripRow\` in \`EncounterUITests/RunnerUITests.swift\` consistently fails on iOS 26 (iPhone 17 Pro simulator). The test cannot find \`runner.player-row\` within the 5-second timeout.

```
RunnerUITests.swift:166: error: XCTAssertTrue failed - Player row should be visible in strip
```

## Reproduction

```bash
xcodebuild test -scheme Encounter \
  -destination 'platform=iOS Simulator,name=iPhone 17 Pro' \
  -testPlan AllTests \
  -only-testing:EncounterUITests/RunnerUITests/testPlayerConditionToggleShowsAndHidesInStripRow \
  -resultBundlePath /tmp/Test.xcresult
```

## Likely cause

\`PlayerStrip\` is rendered via \`.safeAreaInset(edge: .bottom)\` in \`EncounterRunnerView\`. On iOS 26, elements inside \`.safeAreaInset\` may reside in a different layer of the accessibility tree, causing \`app.buttons.matching(identifier: \"runner.player-row\").firstMatch.waitForExistence(timeout: 5)\` to fail even when the strip is visually present.

The fix likely involves querying via the strip container (\`app.otherElements[\"runner.player-strip\"].buttons.matching(identifier: \"runner.player-row\")\`) or using coordinate-based interaction.

## Notes

- Pre-dates branch \`feat/finish-phase-4\`; confirmed failing on main after that PR merged
- The player strip and row are correctly defined in \`PlayerStripRow.swift\` (line 80)
- \`navigateToRunner()\` does add a player to the party before running, so \`session.playerSlots\` is non-empty

--- a/Encounter/Services/PlayerStore.swift
+++ b/Encounter/Services/PlayerStore.swift
@@ -76,6 +76,7 @@ public final class PlayerStore {
   /// Switches the storage directory and resets in-memory state.
   /// Call `load()` afterwards to populate from the new location.
   public func relocate(to newDirectory: URL) {
+    guard newDirectory.path != directory.path else { return }
     directory = newDirectory
     players = []
     party = Party(name: "Active Party")

--- a/Encounter/Views/EncounterRunnerView.swift
+++ b/Encounter/Views/EncounterRunnerView.swift
@@ -43,6 +43,9 @@ struct EncounterRunnerView: View {
       )
     }
     .accessibilityIdentifier("runner.adversary-list")
+    .safeAreaInset(edge: .bottom) {
+      PlayerStrip(session: session)
+    }
     .navigationTitle(session.name)
     #if os(iOS)
       .navigationBarTitleDisplayMode(.inline)
@@ -79,10 +82,6 @@ struct EncounterRunnerView: View {
       .accessibilityIdentifier("runner.reset-confirm-button")
     } message: {
       Text("The current session will be cleared. The encounter definition is not changed.")
-    }
-    .safeAreaInset(edge: .bottom) {
-      PlayerStrip(session: session)
-        .accessibilityIdentifier("runner.player-strip")
     }
   }
 }

